projects/igniteui-angular/src/lib/services/transaction/transaction.ts

```diff
@@ -26,6 +35,11 @@ export interface State {
     type: TransactionType;
 }
 
+export interface IStateUpdateEvent {
+    origin: TransactionEventOrigin;
+    actions?: any;
```

Comment:
you can make this mandatory too and send empty array in `clear` method. For all other places you have the necessary actions.